### ISSUE TITLE
Tests bind EF in a different lifecycle to the other frameworks

### ISSUE THREAD

**Issue by lukemcgregor** [ppanyukov/EntityFrameworkBenchmarks#4]
I'm always a bit skeptical when I see differences of more than a factor 2 when looking at benchmarks. In reality most frameworks are pretty close in performance with the exception of edge cases around specific functionality.

I think that your specific 3-6x slower figure is coming from the way you manage your lifecycle. 

https://github.com/ppanyukov/EntityFrameworkBenchmarks/blob/master/src/EntityFrameworkBenchmarks/Startup.cs#L22-L29

You are using a completely different type of lifecycle (transient vs singleton) for EF and dapper. This means (amongst other things) you are creating a new connection when querying EF and reusing the connection with dapper.

**Comment by ppanyukov** [comment]
I will check it out. Even if that's true, the point stands (somewhat) in that out of the box other frameworks and libraries perform better out of the box without understanding all these complexities such as lifecycle etc.

**Comment by lukemcgregor** [comment]
So your argument is that because you don't understand a framework its bad? Sounds a little flawed to me.

**Comment by ppanyukov** [comment]
Not quite. I guess the original benchmark  intent was "test default behaviours" if I can remember correctly. Obviously lack of knowledge of the tool is not the tool's fault, at the same time the defaults of other frameworks result in a much better results :)
